astropy/io/ascii/html.py

```diff
@@ -83,7 +83,14 @@ def process_lines(self, lines):
                                         'installed to read HTML tables')
 
         if 'parser' not in self.html:
-            soup = BeautifulSoup('\n'.join(lines))
+            try:
+                import warnings
+            except ImportError:
```

Comment:
Agreed, no need to catch an import error for `warnings`.  This import should be at the top of the package (no performance hit since in all likelihood some other module will have imported it already).